tests/console/validate_encrypt.pm

```diff
@@ -19,30 +19,86 @@ use strict;
 use warnings;
 use base "opensusebasetest";
 use testapi;
+use Test::Assert ':all';
+use scheduler 'get_test_data';
+use Data::Dumper;
 
-sub run {
+sub parse_crypttab {
+    my @lines    = split(/\n/, script_output("cat /etc/crypttab"));
+    my $crypttab = {};
+    foreach (@lines) {
+        next if /^\s*#.*$/;
+        if ($_ =~ /(?<name>.+?)\s+(?<encrypted_device>.+?)($|\s+(?<password>.*?)($|\s+(?<options>.*)))/) {
+            $crypttab->{$+{name}} = {
+                encrypted_device => $+{encrypted_device},
+                password         => $+{password},
+                options          => $+{options}};
+        }
+    }
+    return $crypttab;
+}
 
-    record_info('crypttab file', 'Verify whether \'/etc/crypttab\' (encrypted device table) file exists.');
-    assert_script_run('test -f /etc/crypttab', fail_message => 'No /etc/crypttab found!');
+sub parse_cryptsetup_status {
+    my $dev    = shift;
+    my @lines  = split(/\n/, script_output("cryptsetup status $dev"));
+    my $status = {message => shift @lines};
+    foreach (@lines) {
+        if ($_ =~ /\s+(?<param>.*):\s+(?<value>.*)$/) {
+            my ($param, $value) = ($+{param}, $+{value});
+            $param =~ s/ /_/g;
+            $status->{$param} = $value;
+        }
+    }
+    return $status;
+}
 
-    record_info('crypt vol status', 'Verify the crypted volumes are active');
-    my @crypted_volumes = script_output q[cat /etc/crypttab | awk '{print $1}'];
-    foreach (@crypted_volumes) {
-        assert_script_run qq[cryptsetup status $_ | grep "is active"];
+sub verify_crypttab {
+    my $crypttab_expected = shift;
+    record_info("crypttab", "Verify file existence and content");
+    assert_script_run("test -f /etc/crypttab", fail_message => "No /etc/crypttab found");
+    my $crypttab = parse_crypttab();
+    foreach (@{$crypttab_expected->{devices}}) {
```

Comment:
If I correctly understood the piece of code, you're finding a difference between arrays.

Could you please take a look at function `sub arrays_differ` in utils.pm. Maybe it is the one that would fit your needs?

Reply:
I prefer here not traverse the array again (`foreach` + `grep` in `arrays_differ`) and I'm more interested in the error displayed. I tried with a function in `Test::Assert` that does that as well before this approach and was not clear the error neither.